components/accelerator/nvidia/error/xid/helper.go

```diff
@@ -0,0 +1,136 @@
+package xid
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/leptonai/gpud/components"
+	nvidia_query_xid "github.com/leptonai/gpud/components/accelerator/nvidia/query/xid"
+	"github.com/leptonai/gpud/components/common"
+	"github.com/leptonai/gpud/log"
+)
+
+const (
+	StateHealthy   = 0
+	StateDegraded  = 1
+	StateUnhealthy = 2
+)
+
+const rebootThreshold = 2
+
+// EvolveHealthyState resolves the state of the XID error component.
+// note: assume events are sorted by time in ascending order
+func EvolveHealthyState(events []components.Event) (ret components.State) {
+	defer func() {
+		log.Logger.Debugf("EvolveHealthyState: %v", ret)
+	}()
+	var lastSuggestedAction *common.SuggestedActions
+	var lastXidErr *XidError
+	lastHealth := StateHealthy
+	xidRebootMap := make(map[uint64]int)
+	for i := len(events) - 1; i >= 0; i-- {
```

Comment:
It seems that the events are sorted by time in descending order? This conflicts with the comment.

Reply:
yes, discussed with other maintainers offline, we decided its best practice for Event interface to give back descending order events (thus descending here), comment is wrong, updated..